### DISCUSSION TITLE
The document says that we should capture the parameter of a coroutine if its type is ReactPromise or JSValue?

### DISCUSSION THREAD

**Discussion by noelemahcz** [microsoft/react-native-windows#11146] [Q&A] [answered]
It seems that C++20 ensure that parameters of coroutines are kept in the coroutine frame, which means they will live long enough. 
Hence, for parameters passed by value, why should we capture/save them on stack?
I supposed that ReactPromise might contain mechanisms like reference counting, maybe i was wrong?

**Comment by chrisglein** [comment, marked as answer]
Which document are you referring to? This one?
https://microsoft.github.io/react-native-windows/docs/native-modules-async

**Reply by chrisglein** [reply]
At the end of the day if the ReactPromise stays alive through some reference somewhere, that's what you want. If C++20's coroutines have changed so the guidance can be changed, well then we should update it. But that's the intent.
